@claude 

```
17:13:55.896	✓ built in 1m 50s
17:13:55.896	
17:13:55.896	Run npm run preview to preview your production build locally.
17:13:55.896	
17:13:55.896	> Using @sveltejs/adapter-cloudflare
17:13:55.988	▲ [WARNING] Processing wrangler.toml configuration:
17:13:55.988	
17:13:55.988	    - Unexpected fields found in site field: "main"
17:13:55.988	    - Because you've defined a [site] configuration, we're defaulting to "workers-site" for the deprecated `site.entry-point`field.
17:13:55.989	      Add the top level `main` field to your configuration file:
17:13:55.989	      ```
17:13:55.989	      main = "workers-site/index.js"
17:13:55.989	      ```

17:13:56.270	Success: Build command completed
17:13:56.271	Executing user deploy command: npx wrangler versions upload
17:13:57.737	
17:13:57.737	 ⛅️ wrangler 4.35.0
17:13:57.737	───────────────────
17:13:57.795	▲ [WARNING] Processing wrangler.toml configuration:
17:13:57.795	
17:13:57.796	    - Unexpected fields found in site field: "main"
17:13:57.796	    - Because you've defined a [site] configuration, we're defaulting to "workers-site" for the deprecated `site.entry-point`field.
17:13:57.796	      Add the top level `main` field to your configuration file:
17:13:57.796	      ```
17:13:57.796	      main = "workers-site/index.js"
17:13:57.796	      ```

17:13:57.798	✘ [ERROR] The entry-point file at "workers-site/index.js" was not found.
17:13:57.799	
17:13:57.799	
17:13:57.825	🪵  Logs were written to "/opt/buildhome/.config/.wrangler/logs/wrangler-2025-09-11_11-43-57_292.log"
17:13:57.962	Failed: error occurred while running deploy command
```

--- a/package.json
+++ b/package.json
@@ -47,6 +47,7 @@
     "clsx": "^2.1.1",
     "nodemailer": "^7.0.6",
     "tailwind-merge": "^3.3.1",
+    "worker-mailer": "^1.1.5",
     "zod": "^4.1.7"
   }
 }

--- a/src/lib/email.ts
+++ b/src/lib/email.ts
@@ -1,119 +1,153 @@
-import nodemailer from 'nodemailer';
-import { z } from 'zod';
+import { z } from "zod";
 
 export const contactSchema = z.object({
-	name: z.string().min(1, 'Name is required'),
-	email: z.string().email('Invalid email address'),
-	message: z.string().min(1, 'Message is required')
+  name: z.string().min(1, "Name is required"),
+  email: z.email("Invalid email address"),
+  message: z.string().min(1, "Message is required"),
 });
 
 export type ContactData = z.infer<typeof contactSchema>;
 
 export interface EmailConfig {
-	host: string;
-	port: number;
-	secure: boolean;
-	user: string;
-	password: string;
+  host: string;
+  port: number;
+  secure: boolean;
+  user: string;
+  password: string;
 }
 
 export class EmailError extends Error {
-	constructor(public code: string, message: string, public originalError?: any) {
-		super(message);
-		this.name = 'EmailError';
-	}
-}
-
-export function createTransporter(config: EmailConfig) {
-	return nodemailer.createTransport({
-		host: config.host,
-		port: config.port,
-		secure: config.secure,
-		auth: {
-			user: config.user,
-			pass: config.password
-		}
-	});
+  constructor(
+    public code: string,
+    message: string,
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    public originalError?: any,
+  ) {
+    super(message);
+    this.name = "EmailError";
+  }
 }
 
 function sanitizeText(text: string): string {
-	// Basic sanitization to prevent injection attacks
-	return text.replace(/[<>\"'&]/g, (match) => {
-		switch (match) {
-			case '<': return '&lt;';
-			case '>': return '&gt;';
-			case '"': return '&quot;';
-			case "'": return '&#x27;';
-			case '&': return '&amp;';
-			default: return match;
-		}
-	});
+  // Basic sanitization to prevent injection attacks
+  return text.replace(/[<>\"'&]/g, (match) => {
+    switch (match) {
+      case "<":
+        return "&lt;";
+      case ">":
+        return "&gt;";
+      case '"':
+        return "&quot;";
+      case "'":
+        return "&#x27;";
+      case "&":
+        return "&amp;";
+      default:
+        return match;
+    }
+  });
 }
 
-export function createContactEmailOptions(contactData: ContactData, fromEmail: string, toEmail: string) {
-	// Sanitize contact data before using in email
-	const sanitizedData = {
-		name: sanitizeText(contactData.name.trim()),
-		email: contactData.email.trim(), // Email is already validated by Zod
-		message: sanitizeText(contactData.message.trim())
-	};
+export function createContactEmailOptions(
+  contactData: ContactData,
+  fromEmail: string,
+  toEmail: string,
+) {
+  // Sanitize contact data before using in email
+  const sanitizedData = {
+    name: sanitizeText(contactData.name.trim()),
+    email: contactData.email.trim(), // Email is already validated by Zod
+    message: sanitizeText(contactData.message.trim()),
+  };
 
-	return {
-		from: fromEmail,
-		to: toEmail,
-		subject: `New Contact Form Submission from ${sanitizedData.name}`,
-		text: `Name: ${sanitizedData.name}
+  return {
+    from: fromEmail,
+    to: toEmail,
+    subject: `New Contact Form Submission from ${sanitizedData.name}`,
+    text: `Name: ${sanitizedData.name}
 Email: ${sanitizedData.email}
 Message: ${sanitizedData.message}
 
 Sent from: vishalk.com contact form`,
-		replyTo: sanitizedData.email
-	};
+    reply: sanitizedData.email,
+    replyTo: sanitizedData.email,
+  };
 }
 
-export async function sendContactEmail(contactData: ContactData): Promise<{ success: true; message: string }> {
-	const config: EmailConfig = {
-		host: 'smtp.zoho.com',
-		port: 465,
-		secure: true,
-		user: process.env.ZOHO_EMAIL!,
-		password: process.env.ZOHO_PASSWORD!
-	};
+export async function sendContactEmail(
+  contactData: ContactData,
+): Promise<{ success: true; message: string }> {
+  const config: EmailConfig = {
+    host: "smtp.zoho.com",
+    port: 465,
+    secure: true,
+    user: process.env.ZOHO_EMAIL!,
+    password: process.env.ZOHO_PASSWORD!,
+  };
 
-	// Validate environment variables
-	if (!config.user || !config.password) {
-		throw new EmailError('CONFIG_ERROR', 'Email configuration is missing');
-	}
+  // Validate environment variables
+  if (!config.user || !config.password) {
+    throw new EmailError("CONFIG_ERROR", "Email configuration is missing");
+  }
 
-	try {
-		const transporter = createTransporter(config);
-		const mailOptions = createContactEmailOptions(contactData, config.user, config.user);
+  try {
+    const mailOptions = createContactEmailOptions(
+      contactData,
+      config.user,
+      config.user,
+    );
 
-		await transporter.sendMail(mailOptions);
+    if (process.env.NODE_ENV === "development") {
+      const nodemailer = await import("nodemailer");
+      const transporter = nodemailer.createTransport({
+        host: config.host,
+        port: config.port,
+        secure: config.secure,
+        auth: {
+          user: config.user,
+          pass: config.password,
+        },
+      });
+      await transporter.sendMail(mailOptions);
+    } else {
+      const { WorkerMailer } = await import("worker-mailer");
+      const mailer = await WorkerMailer.connect({
+        credentials: {
+          username: config.user,
+          password: config.password,
+        },
+        authType: "plain",
+        host: config.host,
+        port: config.port,
+        secure: config.secure,
+      });
+      await mailer.send(mailOptions);
+    }
 
-		return {
-			success: true,
-			message: 'Message sent successfully!'
-		};
-	} catch (error: any) {
-		// Log error without exposing sensitive information
-		console.error('Email sending error:', {
-			code: error.code,
-			message: error.message,
-			timestamp: new Date().toISOString()
-		});
+    return {
+      success: true,
+      message: "Message sent successfully!",
+    };
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    // Log error without exposing sensitive information
+    console.error("Email sending error:", {
+      code: error.code,
+      message: error.message,
+      timestamp: new Date().toISOString(),
+    });
 
-		// Handle specific nodemailer errors
-		if (error.code === 'EAUTH') {
-			throw new EmailError('EAUTH', 'Email authentication failed');
-		} else if (error.code === 'ENOTFOUND') {
-			throw new EmailError('ENOTFOUND', 'Email server not found');
-		} else if (error.code === 'ECONNECTION') {
-			throw new EmailError('ECONNECTION', 'Failed to connect to email server');
-		} else if (error.code === 'ETIMEDOUT') {
-			throw new EmailError('ETIMEDOUT', 'Email server connection timeout');
-		} else {
-			throw new EmailError('UNKNOWN', 'Failed to send email', error);
-		}
-	}
+    // Handle specific nodemailer errors
+    if (error.code === "EAUTH") {
+      throw new EmailError("EAUTH", "Email authentication failed");
+    } else if (error.code === "ENOTFOUND") {
+      throw new EmailError("ENOTFOUND", "Email server not found");
+    } else if (error.code === "ECONNECTION") {
+      throw new EmailError("ECONNECTION", "Failed to connect to email server");
+    } else if (error.code === "ETIMEDOUT") {
+      throw new EmailError("ETIMEDOUT", "Email server connection timeout");
+    } else {
+      throw new EmailError("UNKNOWN", "Failed to send email", error);
+    }
+  }
 }

--- a/svelte.config.cloudflare.js
+++ b/svelte.config.cloudflare.js
@@ -8,11 +8,12 @@ const config = {
   preprocess: vitePreprocess(),
   kit: {
     adapter: adapter({
-      // Cloudflare Workers configuration for static assets
-      routes: {
-        include: ['/*'],
-        exclude: ['<all>']
-      }
+      // Cloudflare Pages configuration for static site hosting
+      pages: "build",
+      assets: "build",
+      fallback: undefined,
+      precompress: false,
+      strict: true
     })
   }
 };

--- a/svelte.config.cloudflare.js
+++ b/svelte.config.cloudflare.js
@@ -8,14 +8,20 @@ const config = {
   preprocess: vitePreprocess(),
   kit: {
     adapter: adapter({
-      // Cloudflare Pages configuration for static site hosting
-      pages: "build",
-      assets: "build",
-      fallback: undefined,
-      precompress: false,
-      strict: true
-    })
-  }
+      // See below for an explanation of these options
+      config: undefined,
+      platformProxy: {
+        configPath: undefined,
+        environment: undefined,
+        persist: undefined,
+      },
+      fallback: "plaintext",
+      routes: {
+        include: ["/*"],
+        exclude: ["<all>"],
+      },
+    }),
+  },
 };
 
 export default config;

--- a/svelte.config.js
+++ b/svelte.config.js
@@ -8,14 +8,20 @@ const config = {
   preprocess: vitePreprocess(),
   kit: {
     adapter: adapter({
-      // Cloudflare Pages configuration
-      pages: "build",
-      assets: "build",
-      fallback: undefined,
-      precompress: false,
-      strict: true
-    })
-  }
+      // See below for an explanation of these options
+      config: undefined,
+      platformProxy: {
+        configPath: undefined,
+        environment: undefined,
+        persist: undefined,
+      },
+      fallback: "plaintext",
+      routes: {
+        include: ["/*"],
+        exclude: ["<all>"],
+      },
+    }),
+  },
 };
 
 export default config;

--- a/wrangler.toml
+++ b/wrangler.toml
@@ -4,7 +4,3 @@ compatibility_flags = ["nodejs_compat_v2"]
 
 [site]
 bucket = "./build"
-
-# Workers configuration for static site hosting
-# This allows serving static assets through Cloudflare Workers
-main = "./build/_worker.js"

--- a/wrangler.toml
+++ b/wrangler.toml
@@ -1,6 +1,8 @@
 name = "personal"
-compatibility_date = "2025-09-11"
+compatibility_date = "2025-09-01"
 compatibility_flags = ["nodejs_compat_v2"]
+main = ".svelte-kit/cloudflare/_worker.js"
 
-[site]
-bucket = "./build"
+[assets]
+binding = "ASSETS"
+directory = ".svelte-kit/cloudflare"

